src/core/repo.rs

```diff
@@ -26,6 +28,20 @@ impl Target for Repo {
     fn template_context(&self, config: &Config) -> Value {
         repo_context(config, self)
     }
+
+    fn rename(&mut self, new_name: &str) -> std::io::Result<()> {
```

Comment:
One of the patterns I tried to employ here was to keep the `Repo` as a kind of DTO and then use the `tasks::*` stuff to handle any changes to the state of the repo on disk (makes testing and composing functionality a bit easier).

I suspect it might make sense to have a `tasks::MoveDirectory` (or equivalent) which handles this in a re-usable way. That also lets you tie it into the existing `sequence![...]` macro for running a series of migration tasks (as you see in the `fix` command) 😄.